web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -76,6 +91,21 @@
   onMount(async () => {
     document.addEventListener('keydown', onKeyboardPress);
 
+    slideshowStateUnsubscribe = slideshowState.subscribe((value) => {
```

Comment:
So you can do it this way yes, but stores can also be automatically subscribed and unsubscribed by svelte (in a svelte file). You could try doing something like this instead, which I have found works nicely. `$: $slideshowState && onSlideshowStateChange()`, basically use reactivity with the store to trigger code changes instead of manually subscribing and unsubscribing

Reply:
Initially, an automatic subscription/unsubscription seemed a good idea. However, I encountered two issues that I managed to solve only through manual subscription/unsubscription in `onMount`

### Issue 1:
When we call the `Slideshow` command from an album's page (`setAssetId`, followed by `$slideshowState = SlideshowState.PlaySlideshow`) an automatic subscription fires too early because in `handlePlaySlideshow` we need an `assetViewerHtmlElement` to be loaded. As a result we can't enter fullscreen mode. The manual subscription to the `slideshowState` in the `onMount` guarantees that the `assetViewerHtmlElement` is loaded before the subscription is fired.

### Issue 2:
Unexpectedly (at least for me), if I add `$: $slideshowShuffle, handleShuffleChange` then the callback is called on each `setAssetId` call. As a result, the history gets reset on each asset change. This issue was also resolved by using a manual subscription to `slideshowShuffle` in `onMount`.

Maybe there are other ways to solve these issues in svelte but I couldn't find them.

Reply:
OK, sounds good.